### PR TITLE
Update win64 provision scripts (#4151)

### PR DESCRIPTION
**Pull request by momopranto**
Re: #4151 

Provisioning scripts for third party dependencies on Windows are now built using the latest compiler toolset available. Minor refactoring of external executions to use `Start-OsqueryProcess`. Moved `Invoke-VCVarsall` to `osquery_utils.ps1` so it can be used by all other provisioning scripts.

### CODE REVIEW

File: tools/provision/chocolatey/openssl.ps1 [hunk 1]
```diff
@@ -148,4 +148,5 @@ if (Test-Path "$packageName.$chocoVersion.nupkg") {
 else {
   Write-Host "[-] Failed to build $packageName v$chocoVersion." -foregroundcolor Red
 }
-Set-Location $currentLoc
+
```
Comment (muffins) [inline tools/provision/chocolatey/openssl.ps1]: nit: change this back as it's not needed.

File: tools/provision/chocolatey/zstd.ps1 [hunk 2]
```diff
@@ -52,15 +59,29 @@ if(-not (Test-Path $zipFile)) {
...
+  Start-OsqueryProcess $7z $arg $false
 }
+
 Set-Location $sourceDir
 
 $vcxprojLocation = 'build\VS2010\libzstd\libzstd.vcxproj'
 # Patch the AssemblerOutput out of the project
 (Get-Content $vcxprojLocation).replace('<AssemblerOutput>All</AssemblerOutput>', '<AssemblerOutput>NoListing</AssemblerOutput>') | Set-Content $vcxprojLocation
-msbuild "build\VS2010\zstd.sln" /verbosity:minimal /nologo /t:Clean,libzstd /p:Platform=x64 /p:Configuration=Release /p:PlatformToolset=v140
+#msbuild "build\VS2010\zstd.sln" /verbosity:minimal /nologo /t:Clean,libzstd /p:Platform=x64 /p:Configuration=Release /p:PlatformToolset=v141
```
Comment (muffins) [inline tools/provision/chocolatey/zstd.ps1]: Remove this

File: tools/provision/chocolatey/zstd.ps1 [hunk 3]
```diff
@@ -52,15 +59,29 @@ if(-not (Test-Path $zipFile)) {
...
+$msbuildArgs = @(
+	"build\VS2010\zstd.sln",
+	"/verbosity:minimal",
+	"/nologo",
+	"/t:Clean,libzstd",
+	"/p:Platform=x64",
+	"/p:Configuration=Release",
+	"/p:PlatformToolset=v141"
+)
+# Start-OsqueryProcess $msbuild $msbuildArgs $false
```
Comment (muffins) [inline tools/provision/chocolatey/zstd.ps1]: uncomment this

**Comment by muffins** [comment]
ok to test

**Comment by muffins** [comment]
jenkins test freebsd

**Comment by muffins** [comment]
jenkins test FreeBSD

**Comment by muffins** [comment]
@momopranto I think this needs a rebase, any chance you could `git pull --rebase` and push?

**Comment by mike-myers-tob** [comment]
What's the status of this work now, with all of the recent refactoring to the build system? Does it need to be revisited, or has it been overcome by events (OBE)?

**Comment by muffins** [comment]
I think this has been somewhat abandon. Let’s go ahead and close considering how old this is and we can re-open a PR if this is ever revisited.

**Pull request closed without merging**